src/crawlee/beautifulsoup_crawler/_utils.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING
+
+from bs4 import BeautifulSoup, NavigableString, PageElement, Tag
+
+from crawlee._utils.html_to_text import (
+    _ANY_CONSECUTIVE_WHITE_SPACES,
+    _EMPTY_OR_ENDS_WITH_ANY_WHITE_SPACE,
+    _EMPTY_OR_ENDS_WITH_NEW_LINE,
+    BLOCK_TAGS,
+    SKIP_TAGS,
+)
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+
+
+def html_to_text(source: str | BeautifulSoup) -> str:
+    """Converts markup string or `BeautifulSoup` to newline separated plain text without tags using BeautifulSoup."""
```

Comment:
I believe Honza will want to expose this as a normal function as well. In that case, it will be great if you will document the args and return type as well and expose it to the API docs. (Same for the second variant.)

Reply:
I did it in src/crawlee/utils.py . But it is little weird with those implementation variants

Reply:
Oh, I missed that. Please avoid using a general `crawlee.utils` subpackage. Instead, let's place them under the `crawlee.beautifulsoup_crawler` and `crawlee.parsel_crawler` subpackages. See -> https://github.com/apify/crawlee-python/pull/792#discussion_r1886642768.